pkg/ipamd/ipamd.go

```diff
@@ -992,15 +1094,28 @@ func (c *IPAMContext) nodeIPPoolReconcile(interval time.Duration) {
 
 	// Mark phase
 	for _, attachedENI := range attachedENIs {
-		eniIPPool, err := c.dataStore.GetENIIPs(attachedENI.ENIID)
-		if err == nil {
-			// If the attached ENI is in the data store
-			log.Debugf("Reconcile existing ENI %s IP pool", attachedENI.ENIID)
-			// Reconcile IP pool
-			c.eniIPPoolReconcile(eniIPPool, attachedENI, attachedENI.ENIID)
-			// Mark action, remove this ENI from currentENIs map
-			delete(currentENIs, attachedENI.ENIID)
-			continue
+		if !c.enableIpv4PrefixDelegation {
```

Comment:
Don't we need to reconcile both SIP and PDs?  (What if the mode is toggled and we still had some pods running on the 'other' mode?)

Reply:
Yeah, I am handling that in the delete path. After the IP is deleted, I either delete the secondary IP or the prefix based on the mode. But as you suggested, I will fix up the reconciler and remove the cleanup in the delete path.